pkg/florist/user.go

```diff
@@ -70,19 +71,31 @@ func SupplementaryGroups(username string, groups ...string) error {
 // UserSystemAdd adds the system user 'username' and group 'username', with
 // home directory 'homedir' and mode 0o755.
 func UserSystemAdd(username string, homedir string) error {
+	return UserSystemAddWithUID(username, homedir, -1)
+}
+
+// UserSystemAdd adds the system user 'username', group 'username' and force userid to a
+// given number, with home directory 'homedir' and mode 0o755.
+func UserSystemAddWithUID(username string, homedir string, uid int) error {
 	log := Log().With("user", username)
 
 	if _, err := user.Lookup(username); err == nil {
 		log.Debug("user-system-add", "status", "user already present")
 		return nil
 	}
 
-	cmd := exec.Command(
-		"adduser",
-		"--system", "--group",
+	cmdFlags := []string{
+		username,
+		"--system",
+		"--group",
 		"--home", homedir,
 		"--gecos", fmt.Sprintf(`"user %s"`, username),
-		username)
+	}
+	if uid != -1 {
```

Comment:
at the end, this looks quite clean :) at least much cleaner than what we expected (and what you sketched on the piece of paper)